src/odemis/driver/piezomotor.py

```diff
@@ -0,0 +1,733 @@
+                raise PMDError(2, "Encoder error(serial communication or reported error from serial encoder)")
+            elif status[0] == 2:
+                raise PMDError(3, "Supply voltage or motor fault was detected.")
+            elif status[1] == 1:
+                raise PMDError(4, "Command timeout occurred or a syntax error was detected when response was not allowed.")
+            elif status[1] == 8:
+                raise PMDError(5, "Power-on/reset has occurred.")
+
+    def _updatePosition(self):
+        """
+        update the position VA
+        """
+        pos = {}
+        for axname, axis in self._axis_map.items():
+            pos[axname] = self.getEncoderPosition(axis)
+        logging.debug("Reporting new position at %s", pos)
+        self.position._set_value(pos, force_write=True)
+
+    def stopAxis(self, axis):
+        self._sendCommand(b'X%dS' % self._axis_map[axis])
+
+    def getVersion(self, axis):
+        """
+        :returns (str): controller type and firmware version, e.g. 'PMD401 V13'
+        """
+        return self._sendCommand(b'X%d?' % axis)
+
+    def getSerialNumber(self, axis):
+        """
+        :returns (str): serial number
+        """
+        return self._sendCommand(b'X%dY42' % axis)
+
+    def getEncoderPosition(self, axis):
+        """
+        :returns (dict: str --> int): current position of axes as reported by encoders
+        """
+        return int(self._sendCommand(b'X%dE' % axis)) * ENCODER_STEPSIZE
+
+    def setWaveform(self, axis, wf):
+        """
+        :arg wf (waveform): waveform to set
+        """
+        if wf not in (WAVEFORM_DELTA, WAVEFORM_RHOMB, WAVEFORM_PARK):
+            raise ValueError("wf %s not a valid waveform" % wf)
+
+        self._sendCommand(b'X%dM%d' % (axis, wf))
+
+    def getStatus(self, axis):
+        """
+        :returns (str): 4-number status code
+        The most important values are the following
+        First bit:
+        8: communication error (wrong baudrate, data collision, or buffer overflow)
+        4: encoder error (serial communication or reported error from serial encoder)
+        2: voltage error (supply voltage or motor fault was detected)
+        1: command error (command timeout occurred or a syntax error was detected when response was not allowed)
+
+        Second bit:
+        8: reset (power on/ reset occurred)
+        1: index (index signal was detected since last report)
+
+        For all codes, please refer to the PMD-401 manual.
+        """
+        return self._sendCommand(b'X%dU0' % axis)
+
+    def isMovingClosedLoop(self, axis, target):
+        """
+        :param axis (int): axis number
+        :param target (float): target position for axes
+        :returns (bool): True if moving, False otherwise
+        """
+        resp = self._sendCommand(b'X%dE' % axis)
+        if abs(float(target) - float(resp)) < 1e-9:  # 1 nm accuracy
+            return True
+        else:
+            return False
```

Comment:
I don't think you can check that it's moving this way. Firstly the accuracy is far too high (it'll stop more likely before being within 1 nm), but more importantly there are other reason that the move can stop (eg, an error). Looking at the doc, there is no very clear way to check, but I'd say that looking at U0 targetReached will be a good start. You probably also want to check it's still in targetMode (and that targetLimit is not reached). Maybe just looking at the running status is enough. Another possibility again, could be to use Y23.  Most likely, the only way to be really sure is to test.